src/useLocalStore.ts

```diff
@@ -4,14 +4,19 @@ import React from "react"
 import { useAsObservableSourceInternal } from "./useAsObservableSource"
 import { isPlainObject } from "./utils"
 
+export function useLocalStore<TStore extends Record<string, any>>(initializer: () => TStore): TStore
 export function useLocalStore<TStore extends Record<string, any>, TSource extends object = any>(
     initializer: (source: TSource) => TStore,
+    current: TSource
+): TStore
+export function useLocalStore<TStore extends Record<string, any>, TSource extends object = any>(
```

Comment:
I am not sure I follow the reason for this last overload. Doesn't make it accept `undefined`?

Reply:
According to the [docs](https://www.typescriptlang.org/docs/handbook/functions.html#overloads) last "overload" is not a part of overload list, it's what TS calls "implementation" or base case and it must implement all actual overloads

It's not used for actual typing and not exported in typing files, just like base case for `observer()`

While type-checking TS would use first [of two] "compiling" overload and not the base case, so:
- something like `useLocalStore(() => ({}), undefined)` would fail at second overload with `TS2345: Argument of type 'undefined' is not assignable to parameter of type 'object'.`
- something like `useLocalStore(() => ({}))` or `useLocalStore(({}) => ({}), {})` would compile picking first or second overload
- something like `useLocalStore(props => ({}))` would fail with `TS2345: Argument of type '(props: any) => {}' is not assignable to parameter of type '() => {}'.`
- and finally something like 
```ts
const source = { foo: "bar" } as { foo: string } | undefined
useLocalStore(
    props => ({
        props
    }),
    source
)
```
where source might be some object or `undefined` would fail with `TS2345: Argument of type '{ foo: string; } | undefined' is not assignable to parameter of type 'object'.   Type 'undefined' is not assignable to type 'object'.`

Reply:
Cool, I did not know about that. Thanks for the exhaustive explanation.